packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml

```diff
@@ -1146,6 +1158,73 @@ spec:
             content: {{or $proxy $image}}
             encoding: base64
             permissions: "0444"
+  - name: apiserverEncryptionConf
+    definitions:
+      - selector:
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+          kind: KubeadmControlPlaneTemplate
+          matchResources:
+            controlPlane: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/files
+              valueFrom:
+                template: |
+                  - path: /etc/kubernetes/encryption-config.yaml
+                    owner: "root:root"
+                    permissions: "0600"
+                    encoding: base64
+                    content: {{ $apiserverEncryptionConf }}
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs
+              valueFrom:
+                template: |
+                  encryption-provider-config: "/etc/kubernetes/encryption-config.yaml"
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes
+              valueFrom:
+                template: |
+                  - name: encryption-conf
+                    hostPath: /etc/kubernetes/encryption-config.yaml
+                    mountPath: /etc/kubernetes/encryption-config.yaml
+  - name: apiserverEventRateLimitConf
+    definitions:
+      - selector:
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+          kind: KubeadmControlPlaneTemplate
+          matchResources:
+            controlPlane: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs
```

Comment:
Just learnt from other issue we've seen, this will overwrite the whole extraArgs.
If you check this slack https://vmware.slack.com/archives/CEVS7024C/p1670354781024059?thread_ts=1670349930.210859&cid=CEVS7024C you will see example we did

We changed from 
```
        path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer
        valueFrom:
          template: |
            certSANs:
            {{- range .extraDNS }}
            - {{ . }}
            {{- end }}

```
To
```
        path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/certSANs
        valueFrom:
          template: |
            {{- range .extraDNS }}
            - {{ . }}
            {{- end }}
```